pkg/kademlia/kademlia.go

```diff
@@ -272,7 +278,6 @@ func (k *Kademlia) ListenAndServe() error {
 		return err
 	}
 	defer grpcServer.Stop()
```

Comment:
There's no reason this should be deferred.

Reply:
Agreed, although my question is now "why is kademlia.ListenAndServe() never called in our codebase?"

Reply:
Ah, because it shoudn't be called... i.e. the listener should be supplied from outside always. So feel free to remove the whole func.